README.md

```diff
@@ -0,0 +1,117 @@
+# SLD Generator Maven Plugin
+
+## About
+
+Geoserver uses `.sld` files as stylesheets for the layers. These stylesheets specify which features are visible and with which style. These stylesheets are large `.xml` files, and their syntax is cumbersome.
+
+This plugin generates these `.sld` files from a simple `.sld.json` definition. These definitions contain the rules that the `.sld` file should contain.
+
+## Usage
+
+### Install Maven Plugin
+
+Add the following plugin and (adjusted) configuration to the `pom.xml` file.
+```xml
+<plugin>
+  <groupId>nl.aerius</groupId>
+  <artifactId>sld-generator-maven-plugin</artifactId>
+  <version>{sld-generator.version}</version>
+  <configuration>
+    <sourcePath>src/main/sld-rivm</sourcePath>
+    <targetPath>src/main/webapp/data/workspaces/calculator/styles</targetPath>
+    <targetPostfix>_generated</targetPostfix>
+  </configuration>
+  <executions>
+    <execution>
+      <goals>
+        <goal>generate-sld</goal>
+      </goals>
+    </execution>
+  </executions>
+</plugin>
+```
+__Configuration__
+* `sourcePath` path to the folder where `.sld.json`-files are located. The folder is searched recursively for these files.
+* `targetPath` path to the folder where `.sld` files should be written to. Typical the `styles`-folder within the Geoserver workspace.
+* `targetPostfix` _Optional_ postfix which will be added to the `.sld` file names.
+
+
+### Example `.sld.json` file
+
+```json
+{
+  "zoomLevelsFile": "zoom_levels_default.json",
+  "rules": [
+    {
+      "condition": "deposition > 1900 && deposition <= 2200",
+      "fillColor": "CC4C02",
+      "strokeColor": "FFFFFF"
+    },
+    {
+      "condition": "deposition > 2200",
+      "fillColor": "8C2D04",
+      "strokeColor": "FFFFFF"
+    }
+  ]
+}
+```
+
+* `zoomLevelsFile` refers to the zoom level definition file. The zoom level definition is separate because it can be shared between styles. If no zoom level file is defined, no filtering takes place on zoom level.
+* `rules` array containing the rules
+  * `condition` contains the boolean condition to render this style to a feature
+  * `fillColor` color to fill the feature (hexadecimal)
+  * `strokeColor` color to outline the feature (hexadecimal)
+  * `imageUrl` url to an image to draw at the center of the feature. The url is relative to the final `.sld` file, can be a local file in geoserver, or an external image (specify full url)
+  * `customDrawSld` custom XML used in the draw part for this rule in the `.sld` file (for advanced options which can not be specified using above properties)
+  * `customConditionSld` custom XML used in the condition part for this rule in the `.sld` file (for advanced options which can not be specified using above properties)
```

Comment:
Maybe put in a table:


| key | Description |
|--|--|
|`condition` | contains the boolean condition to render this style to a feature |
...